### ISSUE TITLE
`torch.Tensor.pow` returns NaN backward result in multiple PyTorch versions

### ISSUE THREAD

**Issue by ILoveSE** [pytorch/pytorch#95111]
### 🐛 Describe the bug

It is similar to [#89903](https://github.com/pytorch/pytorch/issues/89903).

In Pytorch 1.9.0\1.10.0\1.11.0\1.12.0\1.13.0, when input `x` is negative and `y` is complex, both `torch.pow(x,y)` and `torch.Tensor.pow(x,y)` will return NaN values in backward results, 
This NaN value is risky, because it may affect the subsequent calculation.
However,  the [docs](https://pytorch.org/docs/stable/generated/torch.pow.html) don't mention this interesting behavior, and I wonder if this is a feature of PyTorch's `pow` calculations or a bug that needs to be fixed.

### To Reproduce
```python
import numpy as np
import torch
np.random.seed(0)

x = -np.ones(3)
y = np.random.randn(3) + 1j * np.random.randn(3)
g = np.random.randn(3) + 1j * np.random.randn(3)

print("x: ", x)
print("y: ", y)
print("============================")

device = torch.device("cpu:0")
x1 = torch.tensor(x, device=device, requires_grad=True)
y1 = torch.tensor(y, device=device, requires_grad=True)
g1 = torch.tensor(g, device=device)
def test():
  tmp_result= torch.Tensor.pow(x1, y1)
  # tmp_result= torch.pow(x1, y1)
  return tmp_result
o1 = test()
o1.backward(g1)
print('nan backward: ',y1.grad.detach().cpu().numpy())
```

The NaN result is
```
x:  [-1. -1. -1.]
y:  [1.76405235+2.2408932j  0.40015721+1.86755799j 0.97873798-0.97727788j]
============================
nan backward: [nan+nanj nan+nanj nan+nanj]
```

### Expected behavior

If this is a feature of PyTorch, maybe the documentation can add some corresponding reminders and instructions.

If this is a bug that needs to be improved, hopefully, fix the NaN values.

### Versions

GPU models and configuration: RTX 3090
<details>
  <summary>pytorch 1.9.0</summary>
  <pre><code>
  [pip3] numpy==1.20.2
  [pip3] torch==1.9.0
  [pip3] torchelastic==0.2.0
  [pip3] torchtext==0.10.0
  [pip3] torchvision==0.10.0
  [conda] blas                      1.0                         mkl  
  [conda] cudatoolkit               11.1.74              h6bb024c_0    nvidia
  [conda] ffmpeg                    4.3                  hf484d3e_0    pytorch
  [conda] mkl                       2021.2.0           h06a4308_296  
  [conda] mkl-service               2.3.0            py37h27cfd23_1  
  [conda] mkl_fft                   1.3.0            py37h42c9631_2  
  [conda] mkl_random                1.2.1            py37ha9443f7_2  
  [conda] numpy                     1.20.2           py37h2d18471_0  
  [conda] numpy-base                1.20.2           py37hfae3a4d_0  
  [conda] pytorch                   1.9.0           py3.7_cuda11.1_cudnn8.0.5_0    pytorch
  [conda] torchelastic              0.2.0                    pypi_0    pypi
  [conda] torchtext                 0.10.0                     py37    pytorch
  [conda] torchvision               0.10.0               py37_cu111    pytorch</code></pre>
</details>

<details>
  <summary>pytorch 1.10.0</summary>
  <pre><code>
  [pip3] numpy==1.21.2
  [pip3] torch==1.10.0
  [pip3] torch-tb-profiler==0.4.0
  [pip3] torchelastic==0.2.0
  [pip3] torchtext==0.11.0
  [pip3] torchvision==0.11.0
  [conda] blas                      1.0                         mkl  
  [conda] cudatoolkit               11.3.1               ha36c431_9    nvidia
  [conda] ffmpeg                    4.3                  hf484d3e_0    pytorch
  [conda] mkl                       2021.3.0           h06a4308_520  
  [conda] mkl-service               2.4.0            py37h7f8727e_0  
  [conda] mkl_fft                   1.3.1            py37hd3c417c_0  
  [conda] mkl_random                1.2.2            py37h51133e4_0  
  [conda] numpy                     1.21.2           py37h20f2e39_0  
  [conda] numpy-base                1.21.2           py37h79a1101_0  
  [conda] pytorch                   1.10.0          py3.7_cuda11.3_cudnn8.2.0_0    pytorch
  [conda] pytorch-mutex             1.0                        cuda    pytorch
  [conda] torch-tb-profiler         0.4.0                    pypi_0    pypi
  [conda] torchelastic              0.2.0                    pypi_0    pypi
  [conda] torchtext                 0.11.0                     py37    pytorch
  [conda] torchvision               0.11.0               py37_cu113    pytorch</code></pre>
</details>

<details>
  <summary>pytorch 1.11.0</summary>
  <pre><code>
  [pip3] numpy==1.21.6
  [pip3] pytorch-lightning==1.6.3
  [pip3] torch==1.11.0
  [pip3] torch-tb-profiler==0.4.0
  [pip3] torchaudio==0.11.0
  [pip3] torchmetrics==0.9.0
  [pip3] torchvision==0.12.0
  [conda] blas                      1.0                         mkl  
  [conda] cudatoolkit               11.3.1               h2bc3f7f_2  
  [conda] ffmpeg                    4.3                  hf484d3e_0    pytorch
  [conda] mkl                       2021.4.0           h06a4308_640  
  [conda] mkl-service               2.4.0            py38h7f8727e_0  
  [conda] mkl_fft                   1.3.1            py38hd3c417c_0  
  [conda] mkl_random                1.2.2            py38h51133e4_0  
  [conda] numpy                     1.19.5                   pypi_0    pypi
  [conda] numpy-base                1.21.5           py38hf524024_1  
  [conda] pytorch                   1.11.0          py3.8_cuda11.3_cudnn8.2.0_0    pytorch
  [conda] pytorch-lightning         1.6.3                    pypi_0    pypi
  [conda] pytorch-mutex             1.0                        cuda    pytorch
  [conda] torch                     1.10.0                   pypi_0    pypi
  [conda] torch-tb-profiler         0.4.0                    pypi_0    pypi
  [conda] torchaudio                0.11.0               py38_cu113    pytorch
  [conda] torchmetrics              0.9.0                    pypi_0    pypi
  [conda] torchvision               0.12.0               py38_cu113    pytorch</code></pre>
</details>

<details>
  <summary>pytorch 1.12.0</summary>
  <pre><code>
  [pip3] numpy==1.21.5
  [pip3] torch==1.12.0
  [pip3] torchelastic==0.2.0
  [pip3] torchtext==0.13.0
  [pip3] torchvision==0.13.0
  [conda] blas                      1.0                         mkl  
  [conda] cudatoolkit               11.3.1               ha36c431_9    nvidia
  [conda] ffmpeg                    4.3                  hf484d3e_0    pytorch
  [conda] mkl                       2021.4.0           h06a4308_640  
  [conda] mkl-service               2.4.0            py37h7f8727e_0  
  [conda] mkl_fft                   1.3.1            py37hd3c417c_0  
  [conda] mkl_random                1.2.2            py37h51133e4_0  
  [conda] numpy                     1.21.5           py37he7a7128_2  
  [conda] numpy-base                1.21.5           py37hf524024_2  
  [conda] pytorch                   1.12.0          py3.7_cuda11.3_cudnn8.3.2_0    pytorch
  [conda] pytorch-mutex             1.0                        cuda    pytorch
  [conda] torchelastic              0.2.0                    pypi_0    pypi
  [conda] torchtext                 0.13.0                     py37    pytorch
  [conda] torchvision               0.13.0               py37_cu113    pytorch</code></pre>
</details>

<details>
  <summary>pytorch 1.13.0</summary>
  <pre><code>
  [pip3] numpy==1.22.3
  [pip3] torch==1.13.0
  [pip3] torchtext==0.14.0
  [pip3] torchvision==0.14.0
  [conda] blas                      1.0                         mkl  
  [conda] ffmpeg                    4.3                  hf484d3e_0    pytorch
  [conda] mkl                       2021.4.0           h06a4308_640  
  [conda] mkl-service               2.4.0            py39h7f8727e_0  
  [conda] mkl_fft                   1.3.1            py39hd3c417c_0  
  [conda] mkl_random                1.2.2            py39h51133e4_0  
  [conda] numpy                     1.22.3           py39he7a7128_0  
  [conda] numpy-base                1.22.3           py39hf524024_0  
  [conda] pytorch                   1.13.0          py3.9_cuda11.6_cudnn8.3.2_0    pytorch
  [conda] pytorch-cuda              11.6                 h867d48c_0    pytorch
  [conda] pytorch-mutex             1.0                        cuda    pytorch
  [conda] torchtext                 0.14.0                     py39    pytorch
  [conda] torchvision               0.14.0               py39_cu116    pytorch</code></pre>
</details>

**Comment by ngimel** [comment]
We are apparently taking `log` of a negative number without promoting it to complex. cc @kshitij12345

**Comment by kshitij12345** [comment]
Fixed in #95198

**Comment by ILoveSE** [comment]
> Fixed in #95198

I have seen this pull request! thank you very much for your efforts! :)
